### PR TITLE
ID22311: Added Description, Image, Domain

### PR DESCRIPTION
**Pull request by sviridovartem**

### CODE REVIEW

File: webapplication/aoi/models.py [hunk 1]
```diff
@@ -55,8 +57,11 @@ class Component(models.Model):
 
     name = models.CharField(max_length=200, blank=False, null=False, unique=True, verbose_name='Component name')
     basic_price = models.DecimalField(_("Basic Price (per 1.sq.km)"), max_digits=9, decimal_places=2, default=0)
-    image = models.CharField(max_length=400, verbose_name='Image')
+    image = models.CharField(max_length=400, verbose_name='Docker Image')
     command = models.CharField(max_length=400, blank=True, null=True, verbose_name="Command")
+    description = models.CharField(max_length=400, blank=True, null=True, verbose_name="Description")
+    domains = ArrayField(models.CharField(max_length=15), size=8, blank=True, verbose_name="Domains", null=True)
+    component_image = models.ImageField(upload_to=settings.MEDIA_PATH, blank=True, null=True, verbose_name="Component picture")
```
Comment (adolgarev) [inline webapplication/aoi/models.py:64]: lets call it description_picture

File: webapplication/requirements.txt [hunk 2]
```diff
@@ -8,6 +8,7 @@ django-admin-rangefilter==0.6.3
 drf-yasg==1.20.0
 uWSGI==2.0.18
 coverage==6.3.2
+Pillow==9.5.0
```
Comment (adolgarev) [inline webapplication/requirements.txt:11]: Why is this needed?

Comment (sviridovartem) [reply webapplication/requirements.txt:11]: ERRORS:
aoi.Component.description_picture: (fields.E210) Cannot use ImageField because Pillow is not installed.
        HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "python -m pip install Pillow".
 *  Terminal will be reused by tasks, press any key to close it.

File: webapplication/aoi/models.py (hunk 1 above)
Comment (sviridovartem) [reply webapplication/aoi/models.py]: done

File: webapplication/sip/settings.py [hunk 3]
```diff
@@ -165,6 +165,7 @@
 
 STATIC_URL = '/static/'
 STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+MEDIA_PATH = 'static'
```
Comment (adolgarev) [inline webapplication/sip/settings.py:168]: This should be /media

**Review by adolgarev — COMMENTED** [review]
Please also update API thus FE can read these data

File: webapplication/sip/settings.py (hunk 3 above)
Comment (adolgarev) [reply webapplication/sip/settings.py:168]: And /media should be mapped to the external folder in docker-compose-*.yml (like db volume)

Comment (sviridovartem) [reply webapplication/sip/settings.py]: done
but if I use `/media` then ```
ImageField's 'upload_to' argument must be a relative path, not an absolute path.
        HINT: Remove the leading slash.
```
that's why `media`

File: docker-compose-prod.yml [hunk 4]
```diff
@@ -20,6 +20,7 @@ services:
     volumes:
       - ./webapplication:/code
       - ../data:/data
+      - ../data/media:/code/media
```
Comment (adolgarev) [inline docker-compose-prod.yml:23]: Better not mix ./webapplication -> /code and ../data/media -> /code/media , as it depends on the order of mounting.
Mount ../data/media to /media

File: docker-compose.yml [hunk 5]
```diff
@@ -24,6 +24,7 @@ services:
     volumes:
       - ./webapplication:/code
       - ./data:/data
+      - ./data/media:/code/media
```
Comment (adolgarev) [inline docker-compose.yml:27]: This is not needed

File: webapplication/sip/settings.py [hunk 6]
```diff
@@ -165,6 +165,7 @@
 
 STATIC_URL = '/static/'
 STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+MEDIA_PATH = 'media'
```
Comment (adolgarev) [inline webapplication/sip/settings.py:168]: /media and you'll need to update uwsgi.ini

**Review by adolgarev — COMMENTED** [review]
Migration is missing

File: docker-compose-prod.yml (hunk 4 above)
Comment (sviridovartem) [reply docker-compose-prod.yml:23]: fixed

File: docker-compose.yml (hunk 5 above)
Comment (sviridovartem) [reply docker-compose.yml:27]: fixed but not removed

File: webapplication/sip/settings.py (hunk 6 above)
Comment (sviridovartem) [reply webapplication/sip/settings.py:168]: done

**Comment by sviridovartem** [comment]
> Please also update API thus FE can read these data

checked

**Comment by sviridovartem** [comment]
> Migration is missing

Added

File: webapplication/sip/settings.py [hunk 7]
```diff
@@ -165,6 +165,8 @@
...
+MEDIA_ROOT = "/"
```
Comment (adolgarev) [inline webapplication/sip/settings.py:169]: MEDIA_ROOT = "/media"
COMPONENT_DESCRIPTION_PICTURE_MEDIA_PATH = "description_picture"
MEDIA_URL = "media/"

Comment (sviridovartem) [reply webapplication/sip/settings.py]: done

**Review by adolgarev — APPROVED** [review]
